camayoc/tests/qpc/ui/utils.py

```diff
@@ -187,20 +224,25 @@ def create_source(view, credential_name, source_type, source_name, addresses):
     Button(modal, 'Save').click()
     view.wait_for_element(locator=Locator('//button[text()="Close"]'))
     Button(modal, 'Close', classes=[Button.PRIMARY]).click()
-    time.sleep(0.2)  # wait for window animation to complete
+
+    # Verify that the new row source has been created.
+    view.wait_for_element(locator=Locator(xpath=row_xpath(source_name)))
+    view.element(locator=Locator(xpath=row_xpath(source_name)))
 
 
 def delete_source(view, source_name):
     """Delete a source through the UI."""
-    time.sleep(0.2)  # animation timing wait
+    clear_toasts(view=view)
+    dash = DashboardView(view)
+    dash.nav.select('Sources')
+    time.sleep(0.5)
+    view.wait_for_element(locator=Locator(
+        xpath=(delete_xpath(source_name))))
     GenericLocatorWidget(view, locator=Locator(
-        xpath=f"""//div[text()="{source_name}"]/ancestor::node()[7]
-            //*[contains(@class,"pficon-delete")]""")).click()
-    time.sleep(0.2)
+        xpath=delete_xpath(source_name))).click()
+    time.sleep(0.5)
```

Comment:
I see you having to sleep to wait for UI in quite a few places. To help with refactoring later how about defining a constant in camayoc.constants like `UI_LOAD_TIME` and use it in these sleeps? Or to be more explicative, could define small function in a utils module `wait_for_ui_animation()`

Reply:
good idea. I'll put it into the utils.